.github/workflows/test-helm-chart.yaml

```diff
@@ -0,0 +1,116 @@
+        # k3s-channel: https://update.k3s.io/v1-release/channels
+        include:
+          - k3s-channel: v1.21
+            test: install
+          - k3s-channel: v1.20
+            test: install
+          - k3s-channel: v1.19
+            test: install
+          - k3s-channel: v1.18
+            test: install
+          - k3s-channel: v1.17
+            test: install
+
+          # We run two upgrade tests where we first install an already released
+          # Helm chart version and then upgrades to the version we are now
+          # testing. We test upgrading from the latest main version.
+          #
+          # It can be very useful to see the "Helm diff" step's output from the
+          # latest dev version.
+          - k3s-channel: v1.19
+            test: upgrade
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Starts a k8s cluster with NetworkPolicy enforcement and installs both
+      # kubectl and helm
+      #
+      # ref: https://github.com/jupyterhub/action-k3s-helm/
+      - uses: jupyterhub/action-k3s-helm@v1
+        with:
+          k3s-channel: v1.19
+          metrics-enabled: false
+          traefik-enabled: false
+          docker-enabled: true
+
+      - uses: actions/checkout@v2
+        if: matrix.test == 'upgrade'
+        with:
+          ref: main
+          path: main
+
+      - name: (Upgrade) Install our helm chart from main branch
+        if: matrix.test == 'upgrade'
+        run: |
+          helm upgrade --install posthog main/charts/posthog --timeout 20m -f ci/values.yaml
+
+      - name: "(Upgrade) Install helm diff"
+        if: matrix.test == 'upgrade'
+        run: |
+          helm plugin install https://github.com/databus23/helm-diff
+
+      - name: "(Upgrade) Helm diff ${{ matrix.upgrade-from }} chart with local chart"
+        if: matrix.test == 'upgrade'
+        run: |
+          helm diff upgrade --install posthog charts/posthog -f ci/values.yaml
+
+      - name: "(Upgrade) Await chart install"
+        if: matrix.test == 'upgrade'
+        uses: jupyterhub/action-k8s-await-workloads@v1
+        with:
+          timeout: 900
+          max-restarts: 15
+
+      - name: Install our helm chart
+        run: |
+          helm upgrade --install posthog charts/posthog --timeout 20m -f ci/values.yaml
+
+      # GitHub Action reference: https://github.com/jupyterhub/action-k8s-await-workloads
+      - name: Await all deployments to start
+        uses: jupyterhub/action-k8s-await-workloads@main
+        with:
+          timeout: 900
+          max-restarts: 15
+
+      - name: Setup for ingestion test
+        run: |
+          export WEB_POD=$(kubectl get pods -l role=web -o jsonpath="{.items[].metadata.name}")
+          kubectl exec $WEB_POD -- python manage.py setup_dev --no-data
+
+          echo "POSTHOG_HOSTNAME=$(kubectl get svc posthog -o jsonpath="{.spec.clusterIP}")" >> $GITHUB_ENV
+
+          # Sleep until web service is ready to respond
+          sleep 20
```

Comment:
could we use the pod status (same as load balancer uses to know if to send traffic)?

Reply:
The above actions wait for the deployment to be ready, which should be the same thing but in practice the first few seconds of requests always failed due to address propagation or smth. Since we don't have a reliable way to wait for that - this. Seems to work great in practice.